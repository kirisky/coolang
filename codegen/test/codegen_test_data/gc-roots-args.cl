class A {
 foo:Int;
 --b:B <- new B; -- this messes up the type name when print As allocation since its allocated after A but before As allocation is printed
};

class B {
 foo:Int;
};

class C {
 foo:Int;
};

class D {
 foo:Int;
};

class Main inherits IO {
  arg_roots(a: A, c: C) : Object {{
    out_string("A and C still alive\n\n");
  }};

  main() : Object { {
    arg_roots(new A, new C);
	out_string("A and C should die now\n\n");
	new Object;
  } };
};
