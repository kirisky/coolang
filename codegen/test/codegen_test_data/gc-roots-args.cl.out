Allocated an object of type: Main
Allocated an object of type: A
Allocated an object of type: C
Allocated a string
A and C still alive
Allocated a string
A and C should die now
Freeing an object of type: C
Freeing an object of type: A
Allocated an object of type: Object
